cellbrowser_tools/validateProcessedImages.py

```diff
@@ -367,9 +367,16 @@ def make_rand_features(dataset, count=6):
     return df
 
 
+def get_csv_features(path: str):
+    # replace NaN values with string "NaN"
+    df = pd.read_csv(path)
+    df.fillna("NaN", inplace=True)
```

Comment:
I could be remembering incorrectly but I doubt this is sufficient to screen bad values in a CSV. If it works great but I found it necessary to keep augmenting the list of not accepted values when I processed the A.D. excel files. 
There's the NaN case which you cover, but I think it can also be None depending how the CSV was generated, and then there are values that are nonsensical that possibly were not initialized. 

I would recommend putting in a check that if the data doesn't conform to expected values raise an exception otherwise you can end up with really warped output.

Reply:
This code has already been tested on our CSV data multiple times and it's doing everything it needs to do.  I am not putting any expectation on the values because their ranges totally depend on what each feature is.  This code can't really validate anything other than catastrophic write failures that put garbage in the file.

Reply:
As usual, this will all change the next time we do a data release ;)

Reply:
I get it but this scares me. If you're displaying mean statistics or in my case if you are using the data downstream without sanity checking for reasonable values it can cause downstream code to explode in unpleasant unpredictable ways. If it's just being displayed then I would say whoever provided the data must manually validate the display data once it's processed and ingested. My 2 cents worth.

Reply:
currently the numeric values are used for display on a scatterplot in CFE. The plotting library also builds histograms and it may do other things like averaging, but ultimately it's for display in CFE.  The front end code does handle the NaNs.  What validation do you recommend?
 
(Also of note: this code has already processed the variance data for production but will probably do it another couple of times if changes are needed)

Reply:
My recommendations are checking things like x >= 0, x < Upper bound and if hard values aren't available then 
sample a small part of the data and then use sigma bounds. I still think the authors of the spreadsheet should have to manually check the values at the end.

Reply:
Ah, yeah, I have no info on that and I don't think this code is responsible for flagging unexpected statistical outliers.  The good news is that after this code runs, it's easy to "eyeball" by looking at the actual plots in CFE